app/src/main/java/de/hottenstein/krimirundgang/TourMapActivity.java

```diff
@@ -0,0 +1,108 @@
+    private List<StopInfo> stoplist;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_tour_map);
+        stoplist = createList(10);
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+
+
+    }
+
+    private List<StopInfo> createList(int size) {
+
+        List<StopInfo> result = new ArrayList<StopInfo>();
+        for (int i = 1; i <= size; i++) {
+            float latOffset = rand.nextFloat() * (randMin - randMax) + randMin;
+            float lngOffset = rand.nextFloat() * (randMin - randMax) + randMin;
+            StopInfo si = new StopInfo();
+            si.title = StopInfo.TITLE_PREFIX + i;
+            si.description = StopInfo.DESCRIPTION_PREFIX + i;
+            si.location = new Location(sLocationName);
+            si.location.setLatitude(52.0000 + latOffset);
+            si.location.setLongitude(9.0000 + lngOffset);
+            si.content = i + StopInfo.CONTENT_DUMMY;
+
+            result.add(si);
+        }
+        return result;
+    }
+
+
+    /**
+     * Manipulates the map once available.
+     * This callback is triggered when the map is ready to be used.
+     * This is where we can add markers or lines, add listeners or move the camera. In this case,
+     * we just add a marker near Sydney, Australia.
+     * If Google Play services is not installed on the device, the user will be prompted to install
+     * it inside the SupportMapFragment. This method will only be triggered once the user has
+     * installed Google Play services and returned to the app.
+     */
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        mMap = googleMap;
+        LatLngBounds.Builder builder = new LatLngBounds.Builder();
+
+        // Add a marker at the location of the first stop
+        for (int i = 0; i < stoplist.size(); i++) {
+            double lat = stoplist.get(i).location.getLatitude();
+            double lng = stoplist.get(i).location.getLongitude();
+            LatLng stop = new LatLng(lat, lng);
+            Marker marker =  mMap.addMarker(new MarkerOptions()
+                    .position(stop)
+                    .title(stoplist.get(i).title)
+                    .snippet(stoplist.get(i).description)
+            );
+            StopInfo siTag = stoplist.get(i);
+            marker.setTag(siTag);
+            builder.include(stop);
+        }
+        mMap.setOnInfoWindowClickListener(this);
+        mMap.moveCamera(CameraUpdateFactory.newLatLngBounds(builder.build(), 50));
+
+    }
+
+   @Override
+    public void onInfoWindowClick(Marker marker) {
+        Toast.makeText(this, "Info window clicked",
+                Toast.LENGTH_SHORT).show();
```

Comment:
I think we agreed to remove this toast in favor of a press of the InfoWindow to lead to the corresponding stop.